### PR TITLE
Fix all Dart 2 implicit cast errors

### PR DESCRIPTION
**Pull request by thosakwe**
This will allow us to finally publish the changes from #130.

The tests pass on my machine; if they also pass on Travis, I'll merge this, and then publish `0.3.2` to Pub.

### CODE REVIEW

File: lib/mongo_dart.dart [hunk 1]
```diff
@@ -4,48 +4,77 @@
...
-import "package:collection/collection.dart";
 
 export 'package:bson/bson.dart';
 export 'package:mongo_dart_query/mongo_dart_query.dart';
 
-part 'src/network/packet_converter.dart';
-part 'src/network/connection_manager.dart';
-part 'src/network/connection.dart';
-part 'src/network/mongo_message_transformer.dart';
+part 'src/connection_pool.dart';
```
Comment (Pacane) [inline lib/mongo_dart.dart]: lol what is this new formatting?

Comment (thosakwe) [reply lib/mongo_dart.dart]: I sorted the imports and parts into alphabetical order, IIRC it's part of the Dart style guide.

On second though, there were definitely a few lines I forgot to sort, so it looks weird for sure, haha.

File: lib/src/gridfs/grid_file.dart [hunk 2]
```diff
@@ -32,7 +32,7 @@ class GridFSFile {
       throw new MongoDartError('no md5 stored');
     }
 
-    Completer completer = new Completer();
+    var completer = new Completer<bool>();
```
Comment (Pacane) [inline lib/src/gridfs/grid_file.dart]: Could be final I think

File: lib/src/network/connection_manager.dart [hunk 3]
```diff
@@ -24,7 +24,7 @@ class _ConnectionManager {
     DbCommand isMasterCommand = DbCommand.createIsMasterCommand(db);
     MongoReplyMessage replyMessage = await connection.query(isMasterCommand);
     _log.fine(() => replyMessage.documents[0].toString());
-    var master = replyMessage.documents[0]["ismaster"];
+    var master = replyMessage.documents[0]["ismaster"] == true;
```
Comment (Pacane) [inline lib/src/network/connection_manager.dart]: you can declare it as bool and remove the `== true` part.

Comment (Pacane) [reply lib/src/network/connection_manager.dart]: ideally `final bool master = ...`

Comment (thosakwe) [reply lib/src/network/connection_manager.dart]: Didn't even realize that. In my head, this was a decent trick to coerce the value into a bool

**Pull request merged**